lib/com/codeblock.tsx

```diff
@@ -63,45 +81,124 @@ export class CodeBlock {
       when: (ctx: Context) => {
         if (!ctx.node) return false;
         if (ctx.node.raw.Rel === "Fields") return false;
-        if (ctx.node.parent && ctx.node.parent.hasComponent(Document)) return false;
+        if (ctx.node.parent && ctx.node.parent.hasComponent(Document))
+          return false;
         return true;
       },
       action: (ctx: Context) => {
         const com = new CodeBlock();
-        ctx.node.addComponent(com);
-        ctx.node.changed();
-        workbench.workspace.setExpanded(ctx.path.head, ctx.path.node, true);
-      }
+        if (ctx?.node) {
+          ctx.node.addComponent(com);
+          ctx.node.changed();
+          workbench.workspace.setExpanded(
+            ctx.path.head,
+            ctx.path.node,
+            true
+          );
+        }
+      },
     });
   }
-
 }
 
-
 const CodeEditor = {
-  oncreate({dom, attrs: {path}}) {
+  oncreate(vnode) {
+    const {
+      dom,
+      attrs: { path },
+    } = vnode;
     const snippet = path.node.getComponent(CodeBlock);
+    //@ts-ignore
     dom.jarEditor = new window.CodeJar(dom, (editor) => {
       // highlight.js does not trim old tags,
       // let's do it by this hack.
       editor.textContent = editor.textContent;
+      //@ts-ignore
       window.hljs.highlightBlock(editor);
-      snippet.language = window.hljs.highlightAuto(editor.textContent).language || "";
+      snippet.language =
+        //@ts-ignore
+        window.hljs.highlightAuto(editor.textContent).language || "";
     });
     dom.jarEditor.updateCode(snippet.code);
-    dom.jarEditor.onUpdate(code => {
+    dom.jarEditor.onUpdate((code) => {
       snippet.code = code;
       path.node.changed();
     });
   },
 
-  view({attrs: {workbench, path}}) {
+  view({ attrs: { workbench, path } }) {
     // this cancels the keydown on the outline node
     // so you can use arrow keys normally
     const onkeydown = (e) => e.stopPropagation();
-    
-    return (
-      <div class="code-editor" onkeydown={onkeydown}></div>
-    )
+
+    return <div class="code-editor" onkeydown={onkeydown}></div>;
+  },
+};
+
+class Output {
+  constructor(vnode) {
+    // Initialize output with m.prop() to preserve state across re-renders
+    this.output = vnode.attrs.output || "";
+    this.updateOutput = vnode.attrs.updateOutput || (() => {});
   }
-}
\ No newline at end of file
+
+  oncreate(vnode) {
+    const {
+      dom,
+      attrs: { path },
+    } = vnode;
+
+    const snippet = path.node.getComponent(CodeBlock);
+
+    // Apply syntax highlighting
+    window.hljs.highlightBlock(dom);
+
+    // Find the button element and add event listener
+    const button = dom.querySelector("button");
+    if (button) {
+      button.addEventListener("click", async () => {
```

Comment:
So this onclick should be defined in the view method and added to the button with JSX.